Sources/PaylikeEngine/ViewModel/Engine/PaylikeEngineChecks.swift

```diff
@@ -0,0 +1,34 @@
+extension PaylikeEngine {
+    
+    internal func checkValidState(valid state: EngineState, callerFunc: String) throws {
+        guard self.state == state else {
```

Comment:
Why do we need this?

Reply:
The payment-data-adder and payment-flow functions have this `checkValidState(...)` function to check if it's in an expected state.

Goal:
Control the order of the called engine API functions. The expected calling order is as follows:
1. populate the repository with data-adder functions (waitingForInput state)
2. startPayment (waitingForInput state)
3. continuePaymet after the first webView hint exchange (webViewChallengeStarted state)
4. finishPayment after the TDS user interaction and webView hint exchange (webViewUserInputRequired state)

Reason:
"Catch error the earliest as possible."